css var()로 테마전환, 404 페이지 추가

sungik-choi opened the pull request:
## 기능 추가

- 서버사이드렌더링 문제 때문에 테마 전체를 css var 로 전환
- 404 페이지 추가 (캔버스 그라데이션)
- 메뉴 링크 렌더링 정규표현식 조건에 따라 자동으로 이뤄지도록 수정 (내부/외부 링크)

sungik-choi committed:
Feat : 테마 전환 : ThemeProvider -> css var()

- 서버사이드렌더링에서 테마 깜빡임 문제가 발생했다.
- https://blog.rhostem.com/posts/2020-06-25-dark-mode-for-gatsby-website 여길 참고해서 해결책을 찾았음
- https://dev.to/amaltapalov/how-i-set-dark-mode-for-gatsby-website-4ni0 여기도
- DOM 트리가 완성되기 전에, 스타일시트를 미리 불러와야 깜빡임 현상을 막을 수 있다.
- css-in-js 라이브러리로 작성한 스타일은 앱이 구동한 후 추가되기 때문에 css var() 로 바꿔야만 해결할 수 있다

sungik-choi committed:
Fix : 잘못된 삼항연산자 수정

sungik-choi committed:
Feat : body.dataset 참조하도록 수정

sungik-choi committed:
Feat : useTheme 훅 수정

- 초기 테마를 설정하던 로직은 gatsby-ssr.js 파일로 이동

sungik-choi committed:
Feat : 기존 theme 사용 -> css var() 테마로 모두 변경

sungik-choi committed:
Feat : theme 파일에서 미디어 쿼리 제외 제거

sungik-choi committed:
Refactor : 불필요한 상수 제거

sungik-choi committed:
Feat : 스타일 일부 수정 및 추가, 변수 사용

sungik-choi committed:
Feat : ThemeContext 추가 (스타일 컴포넌트 아님)

- 스타일 컴포넌트의 theme는 미디어쿼리용으로만 사용하므로, theme 상태를 전달할 별도의 ThemeContext 생성
- 기존 컨텍스트를 사용하던 부분들 코드 변경

sungik-choi committed:
Feat : 외부 링크 정규표현식으로 구별해서 렌더링하도록 수정

- github 하드코딩 제거

sungik-choi committed:
Fix : list key 추가

sungik-choi committed:
Feat : three.js 설치

sungik-choi committed:
Feat : 소개 페이지 SEO 추가

sungik-choi committed:
Feat : 404 페이지 추가

- three.js 사용해서 404 텍스트 3d로 띄울 생각
- 배경 투명 적용

sungik-choi committed:
Fix : 오타 수정

sungik-choi committed:
Feat : 404 페이지 튀어다니는 공 초기 코딩 완료

- 엄청 고생했다..
- 파티클 시스템을 함수형 프로그래밍으로 전환하는걸 생각하지 못하겠어서.. 유튜브 예제대로 glowParitcle 컴포넌트를 만들고 상태값에 해당 클래스를 집어넣었다
- 생각해보니 해당 객체가 함수를 가지고 있게 제작해도 똑같이 작동할 거 같은데, 클래스로 작동하는게 보기가 더 편하니 그대로 유지할 예정

sungik-choi committed:
Feat : 404 페이지 제작 및 스타일링 완료

sungik-choi committed:
Feat : 404 페이지 함수 분리 및 useRef 사용

ghost commented:
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report
**sungik-choi**

:tada: Your build was successful! [See the Deploy preview here.](https://sungikchoi-dev.gtsb.io)

## Build Details

[View the build logs here.](https://gatsbyjs.com/dashboard/0a1f03e2-4319-49d8-9562-417cea95901d/sites/47b582be-70ef-4f85-a003-16aaaf764a14/builds/064afac9-93f0-42c3-8520-ec1f721e4bc0/details)

:clock1: Build time: **22s**

## Performance

### Lighthouse report

| Metric | Score |
| ----- | ---- |
| Performance |   :green_heart: 100   |
| Accessibility |   :green_heart: 98   |
| Best Practices |   :green_heart: 100   |
| SEO |   :large_orange_diamond: 83   |

:link: [View full report](https://build-064afac9-93f0-42c3-8520-ec1f721e4bc0.gtsb.io/reports/lighthouse/index.html)

sungik-choi committed:
Refactor : 불필요한 쿼리문 삭제

sungik-choi committed:
Fix : 404 페이지 useEffect -> useLayoutEffect 변경

- 초기 깜빡임 현상 해결
- 주석 삭제

sungik-choi committed:
Fix : footer 위치 재이동

- 하단에 제대로 위치 안함

sungik-choi committed:
Feat : about 페이지 마크다운 파일로 렌더링 하도록 기능 추가

- posts 디렉토리에 about.md 파일 추가
- 디렉토리 꼬임을 방지하기 위해 blog 디렉토리 아래에 기존 파일들 추가
- 쿼리문 가져올 때 /blog/ 포함된 파일만 가져와서 포스팅 템플릿으로 만듬
- about 페이지에서 useStaticQuery를 통해 about.md 파일만 필터링해서 가져오고 렌더링

sungik-choi committed:
Feat : 소개 페이지 스타일링 및 footer 레포지토리 추가

sungik-choi committed:
Feat : 스타일링 수정

- 아이폰5 사이즈에서 일부 스타일 깨지는 부분 수정

sungik-choi committed:
Feat : manifest 수정 및 파비콘 변경
